### ISSUE TITLE
1-based index

### ISSUE THREAD

**Issue by mitjap** [pglira/simpleICP#24]
I've noticed that in some places you used 1-based indices. I guess all of this should be 0-based? 

https://github.com/pglira/simpleICP/blob/1478851961c01eab677bd325f93dff9b85d7a069/c%2B%2B/src/simpleicp.cpp#L164
https://github.com/pglira/simpleICP/blob/1478851961c01eab677bd325f93dff9b85d7a069/c%2B%2B/src/simpleicp.cpp#L181
https://github.com/pglira/simpleICP/blob/1478851961c01eab677bd325f93dff9b85d7a069/c%2B%2B/src/pointcloud.cpp#L82

**Comment by pglira** [comment]
You are right, all three are real bugs. The loops started at i=1 and skipped the first element.

For Median, vv[0] stayed at 0 (zero-init from std::vector), which shifted the median.

For MAD it was worse: Eigen::VectorXd does not zero-init, so dmed[0] was garbage from the stack. This value goes into the outlier rejection threshold (sigmad = 1.4826 * MAD), so it could affect how many correspondences are rejected.

For SelectNPts, point 0 stayed always active.

Fixed in 1929faa. Thanks for reporting.
